### PR TITLE
Fix OGMail searchable text extender.

### LINKED ISSUE #2462: Unicode Fehler bei E-Mails mit Emojis im Subject
Fehler entstand beim Import des folgenden Mails:

[wg-letzte-chance-23f0-d83cdfc3-sicher-dir-deinen.eml.zip](https://github.com/4teamwork/opengever.core/files/699016/wg-letzte-chance-23f0-d83cdfc3-sicher-dir-deinen.eml.zip)

Danach habe ich das Mail in den Papierkorb gelegt.

https://lab.onegovgever.ch/ordnungssystem/fuehrung/kommissionen-mit-selbstaendigen-verwaltungsbefugnissen/kommission-c/ressorts/dossier-479/wg-letzte-chance-23f0-d83cdfc3-sicher-dir-deinen#overview
N
Zeit
11.01.2017 14:44
Benutzername
sandro.ackermann (sandro.ackermann)
Anfrage URL
https://lab.onegovgever.ch/ordnungssystem/fuehrung/kommissionen-mit-selbstaendigen-verwaltungsbefugnissen/kommission-c/ressorts/dossier-479/trashed
Fehlertyp
ValueError
Fehlerwert
unichr() arg not in range(0x10000) (narrow Python build)
Traceback (innermost last):

Module ZPublisher.Publish, line 70, in publish
Module ZPublisher.mapply, line 77, in mapply
Module ZPublisher.Publish, line 48, in call_object
Module opengever.trash.trash, line 116, in __call__
Module opengever.trash.trash, line 61, in trash
Module Products.CMFCore.CMFCatalogAware, line 88, in reindexObject
Module Products.CMFCore.CatalogTool, line 301, in reindexObject
Module Products.CMFPlone.CatalogTool, line 353, in catalog_object
Module Products.ZCatalog.ZCatalog, line 476, in catalog_object
Module Products.ZCatalog.Catalog, line 339, in catalogObject
Module Products.ZCTextIndex.ZCTextIndex, line 180, in index_object
Module plone.indexer.wrapper, line 65, in __getattr__
Module plone.indexer.delegate, line 18, in __call__
Module collective.dexteritytextindexer.indexer, line 89, in dynamic_searchable_text_indexer
Module ftw.mail.mail, line 183, in __call__
Module ftw.mail.mail, line 144, in convert_to_text
Module ftw.mail.mail, line 171, in convert_to_text
Module Products.PortalTransforms.TransformEngine, line 173, in convertTo
Module Products.PortalTransforms.Transform, line 198, in convert
Module Products.PortalTransforms.libtransforms.retransform, line 30, in convert
Module Products.PortalTransforms.transforms.html_to_text, line 18, in sub_func
ValueError: unichr() arg not in range(0x10000) (narrow Python build)

### PR DESCRIPTION
**Pull request by elioschmutz**
This PR fixes the unicode-issue due to 

- unconfiguring the ftw.mail searchableTextExtender which uses the html_to_text transform
- registering the document searchableTextExtender for IBaseDocument instead the more specific `IDocumentSchema`
- Removing the `OGMail` searchableTextExtender because it's the same implementation like the one of the document.

Before:
screen1

After:
screen1

Closes #2462

### CODE REVIEW

File: opengever/document/indexers.py [hunk 1]
```diff
@@ -36,16 +37,17 @@ def __init__(self, context):
         self.context = context
 
     def extract_text(self):
-        if self.context.file:
-            filename = self.context.file.filename
+        file_ = self.context.get_file()
+        if file_:
+            filename = file_.filename
             transform_tool = getToolByName(self.context, 'portal_transforms')
```
Comment (phgross) [inline opengever/document/indexers.py]: if you touch it anyways please change to plone.api.

Comment (elioschmutz) [reply opengever/document/indexers.py]: @phgross  What should I change to plone.api? the `getToolByName`?

Comment (phgross) [reply opengever/document/indexers.py]: @elioschmutz yes, please (Makes it easier to upgrade plone etc.)

**Comment by elioschmutz** [comment]
@phgross 👍 updated

**Pull request merged**